src/main/java/rodain/core/CreateBagits.java

```diff
@@ -0,0 +1,170 @@
+package rodain.core;
+
+import static java.nio.file.StandardCopyOption.COPY_ATTRIBUTES;
+
+import java.io.*;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.stream.StreamResult;
+import javax.xml.transform.stream.StreamSource;
+
+import org.slf4j.LoggerFactory;
+
+import rodain.rules.TreeNode;
+import rodain.rules.sip.SipPreview;
+import rodain.utils.Utils;
+import gov.loc.repository.bagit.Bag;
+import gov.loc.repository.bagit.BagFactory;
+import gov.loc.repository.bagit.PreBag;
+import gov.loc.repository.bagit.writer.impl.FileSystemWriter;
+
+/**
+ * Created by adrapereira on 06-10-2015.
+ */
+public class CreateBagits extends Thread {
+    private static final org.slf4j.Logger log = LoggerFactory.getLogger(CreateBagits.class.getName());
+    private int successful = 0, error = 0;
+    private String startPath;
+    private long time;
+
+    public CreateBagits(String path){
+        startPath = path;
+    }
+
+    @Override
+    public void run(){
+        long start = System.currentTimeMillis();
+        long lastUpdate = 0;
+        Map<SipPreview, String> previews = Main.getSipPreviews();
+        for(SipPreview preview: previews.keySet()){
+            createBagit(previews.get(preview), preview);
+
+            if(System.currentTimeMillis() - lastUpdate > 1000){
+                Footer.setStatus("Processing... " + successful + " created");
+                lastUpdate = System.currentTimeMillis();
+            }
+        }
+        long end = System.currentTimeMillis();
+        time = end - start;
+        updateFooter();
+    }
+
+    private void updateFooter(){
+        Footer.activeButton();
+        long second = (time / 1000) % 60;
+        Footer.setStatus("Created " + successful + " Bagits. Errors creating " + error + " Bagits. Time: " + second + " seconds");
+    }
+
+    private void createBagit(String schemaId, SipPreview sip){
+        String path = startPath + "/";
+        File ruleDir = new File(path);
+        ruleDir.mkdir();
+        String name = path + sip.getName();
+        //make the directory
+        new File(name).mkdir();
+        new File(name+"/data/").mkdir();
```

Comment:
Use constants instead of magic strings.
Use NIO2 (Paths) instead of File.
Use Path.resolve() or similar method, so the path separator is not hardcoded (this code may not be compatible with Windows OS)